pkgs/by-name/st/stardust-xr-server/package.nix

```diff
@@ -34,28 +41,88 @@ rustPlatform.buildRustPackage rec {
     };
   };
 
+  # CPM sucks soo much
+  buildFeatures = [ "local_deps" ];
+  env = {
+    FORCE_LOCAL_DEPS = true;
+    CPM_LOCAL_PACKAGES_ONLY = true;
+    CPM_SOURCE_CACHE = "./build";
+    CPM_USE_LOCAL_PACKAGES = true;
+    CPM_DOWNLOAD_ALL = false;
+
+    DEP_MESHOPTIMIZER_SOURCE = fetchFromGitHub {
+      name = "meshoptimizer-source";
+      owner = "zeux";
+      repo = "meshoptimizer";
+      rev = "c21d3be6ddf627f8ca852ba4b6db9903b0557858";
+      hash = "sha256-QCxpM2g8WtYSZHkBzLTJNQ/oHb5j/n9rjaVmZJcCZIA=";
+    };
+    DEP_BASIS_UNIVERSAL_SOURCE = fetchFromGitHub {
+      name = "basis_universal-source";
+      owner = "BinomialLLC";
+      repo = "basis_universal";
+      rev = "900e40fb5d2502927360fe2f31762bdbb624455f";
+      hash = "sha256-zBRAXgG5Fi6+5uPQCI/RCGatY6O4ELuYBoKrPNn4K+8=";
+    };
+    DEP_SK_GPU_SOURCE = stdenvNoCC.mkDerivation {
+      name = "sk_gpu-source";
+
+      src = fetchzip {
+        name = "sk_gpu-source-raw";
+        url = "https://github.com/StereoKit/sk_gpu/releases/download/v2024.8.16/sk_gpu.v2024.8.16.zip";
+        stripRoot = false;
+        hash = "sha256-NeFvmCP2JII9jZfO5fj0yAWXutXAU2H6levTpyC5rpo=";
+      };
+
+      postPatch = ''
+        chmod -R 755 tools/*
+      '';
+
+      dontConfigure = true;
+      dontBuild = true;
+
+      nativeBuildInputs = [
+        autoPatchelfHook
+      ];
+
+      installPhase = ''
+        runHook preInstall
+
+        mkdir -p $out
+        cp -R . $out
+
+        runHook postInstall
+      '';
+    };
+
+    LIBCLANG_PATH = "${libclang.lib}/lib";
+  };
+
+  postPatch = ''
+    sk=$(echo $cargoDepsCopy/stereokit-rust-*/StereoKit)
+    install -D ${cpm-cmake}/share/cpm/CPM.cmake $sk/build/cpm/CPM_0.38.7.cmake
+    ln -s $DEP_SK_GPU_SOURCE $sk/sk_gpu
+  '';
+
   nativeBuildInputs = [
     cmake
     pkg-config
-    rustPlatform.bindgenHook
+    llvmPackages.libcxxClang
   ];
 
   buildInputs = [
     fontconfig
     libGL
+    libXau
+    libXdmcp
     libxkbcommon
     mesa
-    openxr-loader
-    xorg.libX11
+    openxr-loader.dev
+    xorg.libX11.dev
```

Comment:
`.dev` isn't needed in mkDerivation inputs, as it is always included (if present)